pkg/controller/service/service_controller.go

```diff
@@ -317,12 +317,24 @@ func (s *ServiceController) syncLoadBalancerIfNeeded(key string, service *v1.Ser
 
 		// Update the status on the copy
 		service.Status.LoadBalancer = *newState
-
+		maxRetries := 3
 		if err := s.persistUpdate(service); err != nil {
```

Comment:
If you read further, `persistUpdate()` already implemented such retry logic.

Reply:
Thanks, I haven't noticed that.
I'll close the PR.